### PR TITLE
IGNITE-15103 Implement debug and error logging of connections and queries

### PR DESCRIPTION
**Pull request by ivandasch**

### CODE REVIEW

File: pyignite/connection/connection.py [hunk 1]
```diff
@@ -28,6 +29,8 @@
 
 CLIENT_STATUS_AUTH_FAILURE = 2000
 
+logger = logging.getLogger('.'.join(__name__.split('.')[:-1]))
```
Comment (isapego) [inline pyignite/connection/connection.py:32]: Is it a good idea to make it global?

File: pyignite/connection/connection.py [hunk 2]
```diff
@@ -78,21 +81,53 @@ def protocol_context(self):
...
+                         self.host, self.port, self.uuid, self.client.protocol_context)
+
+    def _on_handshake_fail(self, err):
+        if isinstance(err, AuthenticationError):
+            logger.error("Authentication failed while connecting to node(address=%s, port=%d): %s",
+                         self.host, self.port, err)
+        if isinstance(err, connection_errors):
+            logger.error("Failed to perform handshake, connection to node(address=%s, port=%d) "
+                         "with protocol context %s failed: %s",
+                         self.host, self.port, self.client.protocol_context, err, exc_info=True)
```
Comment (isapego) [inline pyignite/connection/connection.py:121]: What about other kind of errors? I believe, we should log them too.

File: pyignite/connection/connection.py [hunk 3]
```diff
@@ -78,21 +81,53 @@ def protocol_context(self):
...
+
+    def _on_connection_lost(self, err=None, expected=False):
+        if expected and logger.isEnabledFor(logging.DEBUG):
+            logger.debug("Connection closed to node(address=%s, port=%d, node_uuid=%s)",
+                         self.host, self.port, self.uuid)
+        else:
+            logger.error("Connection lost to node(address=%s, port=%d, node_uuid=%s): %s",
```
Comment (isapego) [inline pyignite/connection/connection.py:128]: Lost connection to the node is not an error. It should be `warning` or `info` I believe. Client can continue working on lost connection and user should expect that such kind of event may occur.

**Review by isapego — COMMENTED** [review]
Overall looks good. Please, find my comments inline

File: pyignite/connection/connection.py (hunk 1 above)
Comment (ivandasch) [reply pyignite/connection/connection.py:32]: Yep, it is common approach.
For example, [see aiohttp](https://github.com/aio-libs/aiohttp/blob/09ac1cbb4c07b4b3e1872374808512c9bc52f2ab/aiohttp/log.py)

File: pyignite/connection/connection.py (hunk 2 above)
Comment (ivandasch) [reply pyignite/connection/connection.py:121]: Good point, let's catch in this branch all others exceptions, other than `AuthenticationError`

File: pyignite/connection/connection.py (hunk 3 above)
Comment (ivandasch) [reply pyignite/connection/connection.py:128]: Yep, aggree, `info` is enough

**Review by isapego — APPROVED** [review]
Looks good to me.

**Pull request closed without merging**